src/indexer.c

```diff
@@ -959,8 +955,8 @@ int git_indexer_commit(git_indexer *idx, git_transfer_progress *stats)
 	git_oid_fromraw(&file_hash, packfile_trailer);
 	git_mwindow_close(&w);
 
-	git_hash_final(&trailer_hash, &idx->trailer);
-	if (git_oid_cmp(&file_hash, &trailer_hash)) {
+	git_hash_final(&idx->hash, &idx->trailer);
+	if (git_oid_cmp(&file_hash, &idx->hash)) {
```

Comment:
The change would be more obviously correct if these files were kept unchanged and then replacing the `git_hash_final(&idx->hash, &ctx)` with a `git_oid_cpy(&idx->hash, &trailer_hash)`.